lib/logflare/alerting.ex

```diff
@@ -143,22 +148,14 @@ defmodule Logflare.Alerting do
   end
 
   @doc """
-  Retrieves a Citrine.Job based on AlertQuery.
-  Citrine.Job shares the same id as AlertQuery, resulting in a 1-1 relationship.
+  Retrieves a Job based on AlertQuery.
+  Job shares the same id as AlertQuery, resulting in a 1-1 relationship.
   """
   @spec get_alert_job(AlertQuery.t()) :: Citrine.Job.t()
   def get_alert_job(%AlertQuery{id: id}), do: get_alert_job(id)
 
   def get_alert_job(id) do
-    AlertsScheduler.jobs()
-    |> Enum.find_value(fn {_ref,
-                           %Quantum.Job{
-                             task: {_module, _function, [%AlertQuery{id: query_id}, _]}
-                           } = job} ->
-      if query_id == id do
-        job
-      end
-    end)
+    remote_call_scheduler(:find_job, [Integer.to_string(id)])
```

Comment:
Now that you have `remote_call_scheduler`, I would just use that instead of `on_scheduler_node` for consistency, so it's a single approach. Also, this way, if someone needs to change those functions in the future, they don't end up doing multiple RPCs.